src/Zigzag_persistence/include/gudhi/Filtered_zigzag_persistence.h

```diff
@@ -0,0 +1,553 @@
+ *
+ * @brief Default options for @ref Filtered_zigzag_persistence_with_storage and @ref Filtered_zigzag_persistence.
+ */
+struct Default_filtered_zigzag_options {
+  using internal_key = int;        /**< Face ID used internaly, must be signed. */
+  using face_key = int;            /**< Face ID used in the given boundaries. */
+  using filtration_value = double; /**< Filtration value type. */
+  using dimension_type = int;      /**< Dimension value type. */
+  /**
+   * @brief Column type use by the internal matrix.
+   */
+  static const Gudhi::persistence_matrix::Column_types column_type =
+      Gudhi::persistence_matrix::Column_types::INTRUSIVE_LIST;
+};
+
+/**
+ * @ingroup zigzag_persistence
+ *
+ * @brief Class computating the zigzag persistent homology of a zigzag filtration. Algorithm based on \cite zigzag.
+ * Eventhough the insertions and removals are given in a "stream-like" way, the barcode and other values are
+ * stored during the whole process and not removed. It is therefore suited for smaller filtrations where the clean
+ * ups produce a higher overhead than the memory consumption.
+ * 
+ * @tparam FilteredZigzagOptions Structure following the @ref FilteredZigzagOptions concept.
+ * Default value: @ref Default_filtered_zigzag_options.
+ */
+template <class FilteredZigzagOptions = Default_filtered_zigzag_options>
+class Filtered_zigzag_persistence_with_storage
+{
+ public:
+  using Options = FilteredZigzagOptions;                        /**< Zigzag options. */
+  using internal_key = typename Options::internal_key;          /**< Key and index type, has to be signed. */
+  using face_key = typename Options::face_key;                  /**< Face ID type from external inputs. */
+  using filtration_value = typename Options::filtration_value;  /**< Type for filtration values. */
+  using dimension_type = typename Options::dimension_type;      /**< Type for dimension values. */
+  using Index_interval = Interval<internal_key>;                /**< Persistence interval type. */
+
+  /** \brief Structure to store persistence intervals by their filtration values.
+   *
+   * \details By convention, interval \f$[b;d]\f$ are
+   * closed for finite indices b and d, and open for left-infinite and/or
+   * right-infinite endpoints.
+   */
+  struct Filtration_value_interval : Interval<filtration_value> {
+   private:
+    using Base = Interval<filtration_value>;
+
+   public:
+    /**
+     * @brief Default constructor
+     */
+    Filtration_value_interval() : Base() {}
+    /**
+     * @brief Construct a new interval with given parameters
+     *
+     * @param dim Dimension of the interval.
+     * @param b Start value of the interval.
+     * @param d End value of the interval.
+     */
+    Filtration_value_interval(int dim, filtration_value b, filtration_value d) : Base(dim, b, d) {}
+
+    /**
+     * @brief Returns the absolute length of the interval \f$|d-b|\f$.
+     */
+    filtration_value length() const {
```

Comment:
Couldn't we put that in the parent class directly?

Reply:
Honestly, I was thinking of just removing them. They are used nowhere. Or do you think that they can be useful enough for users to keep them? If yes, I would indeed move them in the parent class.